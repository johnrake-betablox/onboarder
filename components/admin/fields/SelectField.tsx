import { DMMF } from "@prisma/client/runtime";
import { ChangeEvent, useEffect } from "react";
import FieldLabel from "../FieldLabel";
import { SelectOption } from "@/lib/admin";

interface Props {
  field: DMMF.Field;
  value: string;
  options: SelectOption[];
  onChange: (key: string, value: string) => void;
}
export default function SelectField({
  field,
  value,
  options = [],
  onChange,
}: Props) {
  const handleChange = (e: ChangeEvent<HTMLSelectElement>) => {
    onChange(field.name, e.currentTarget.value);
  };

  return (
    <div>
      <FieldLabel field={field} />
      <select
        className="focus:shadow-outline w-full rounded border px-3 py-2 leading-tight text-gray-700 shadow"
        id={field.name}
        name={field.name}
        value={value}
        required={field.isRequired}
        onChange={handleChange}
      >
        <option value="" />
        {options.map((option) => (
          <option key={option.value} value={option.value}>
            {option.label}
          </option>
        ))}
      </select>
    </div>
  );
}
