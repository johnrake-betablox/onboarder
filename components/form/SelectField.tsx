type Option = { value: string; label: string };

interface Props {
  field: string;
  value: any;
  required?: boolean;
  options: { value: string; label: string }[];
  onChange: (value: string) => void;
}
export default function SelectField({
  field,
  value,
  onChange,
  options = [],
  required = false,
}: Props) {
  return (
    <select
      id={field}
      name={field}
      required={required}
      className="mt-1 block w-full rounded-md border border-gray-300 px-3 py-2 placeholder-gray-400 shadow-sm focus:border-green-600 focus:outline-none focus:ring-green-600 sm:text-sm"
      value={value}
      onChange={(e) => onChange(e.currentTarget.value)}
    >
      <option value="" />
      {options.map((option) => (
        <option key={option.value} value={option.value}>
          {option.label}
        </option>
      ))}
    </select>
  );
}
