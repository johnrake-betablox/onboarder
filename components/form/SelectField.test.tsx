import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import SelectField from "./SelectField";

const onChange = (value: any) => {};
const options = [
  { value: "KS", label: "Kansas" },
  { value: "CA", label: "California" },
  { value: "TX", label: "Texas" },
];

test("renders a snapshot", () => {
  const { asFragment } = render(
    <SelectField
      field="state"
      value="KS"
      onChange={onChange}
      options={options}
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});

test("renders a required field", () => {
  const { asFragment } = render(
    <SelectField
      field="state"
      value="CA"
      onChange={onChange}
      options={options}
      required
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});
