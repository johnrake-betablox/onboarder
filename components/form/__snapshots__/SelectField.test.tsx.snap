// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a required field 1`] = `
<DocumentFragment>
  <select
    class="mt-1 block w-full rounded-md border border-gray-300 px-3 py-2 placeholder-gray-400 shadow-sm focus:border-green-600 focus:outline-none focus:ring-green-600 sm:text-sm"
    id="state"
    name="state"
    required=""
  >
    <option
      value=""
    />
    <option
      value="KS"
    >
      Kansas
    </option>
    <option
      value="CA"
    >
      California
    </option>
    <option
      value="TX"
    >
      Texas
    </option>
  </select>
</DocumentFragment>
`;

exports[`renders a snapshot 1`] = `
<DocumentFragment>
  <select
    class="mt-1 block w-full rounded-md border border-gray-300 px-3 py-2 placeholder-gray-400 shadow-sm focus:border-green-600 focus:outline-none focus:ring-green-600 sm:text-sm"
    id="state"
    name="state"
  >
    <option
      value=""
    />
    <option
      value="KS"
    >
      Kansas
    </option>
    <option
      value="CA"
    >
      California
    </option>
    <option
      value="TX"
    >
      Texas
    </option>
  </select>
</DocumentFragment>
`;
